tests/components/arcam_fmj/test_config_flow.py

```diff
@@ -33,16 +33,21 @@
 
 MOCK_UPNP_LOCATION = f"http://{MOCK_HOST}:8080/dd.xml"
 
-MOCK_DISCOVER = {
-    ssdp.ATTR_UPNP_MANUFACTURER: "ARCAM",
-    ssdp.ATTR_UPNP_MODEL_NAME: " ",
-    ssdp.ATTR_UPNP_MODEL_NUMBER: "AVR450, AVR750",
-    ssdp.ATTR_UPNP_FRIENDLY_NAME: f"Arcam media client {MOCK_UUID}",
-    ssdp.ATTR_UPNP_SERIAL: "12343",
-    ssdp.ATTR_SSDP_LOCATION: f"http://{MOCK_HOST}:8080/dd.xml",
-    ssdp.ATTR_UPNP_UDN: MOCK_UDN,
-    ssdp.ATTR_UPNP_DEVICE_TYPE: "urn:schemas-upnp-org:device:MediaRenderer:1",
-}
+MOCK_DISCOVER = ssdp.SsdpServiceInfo(
+    ssdp_usn="mock_usn",
+    ssdp_st="mock_st",
+    ssdp_location=f"http://{MOCK_HOST}:8080/dd.xml",
+    upnp={
+        ssdp.ATTR_UPNP_MANUFACTURER: "ARCAM",
+        ssdp.ATTR_UPNP_MODEL_NAME: " ",
+        ssdp.ATTR_UPNP_MODEL_NUMBER: "AVR450, AVR750",
+        ssdp.ATTR_UPNP_FRIENDLY_NAME: f"Arcam media client {MOCK_UUID}",
+        ssdp.ATTR_UPNP_SERIAL: "12343",
+        ssdp.ATTR_UPNP_UDN: MOCK_UDN,
+        ssdp.ATTR_UPNP_DEVICE_TYPE: "urn:schemas-upnp-org:device:MediaRenderer:1",
+    },
+    x_homeassistant_matching_domains=set(),
```

Comment:
Not required if we add a default to in `_HaServiceDescription`.
```suggestion
```

Reply:
This should not be required to mock

Reply:
So do we put in a default in the dataclass (like the suggestions above)?
Or is there a better to keep it as compulsory and yet not need to add in the tests?